src/main/java/ToDoApplication/DAO/BaseDao.java

```diff
@@ -0,0 +1,52 @@
+package ToDoApplication.DAO;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+abstract class BaseDao {
+
+
+    protected void closeResource(Connection connection, PreparedStatement preparedStatement) {
+        if (connection != null) {
+            try {
+                connection.close();
+            } catch (SQLException e) {
+                e.printStackTrace();
+            }
+        }
+        if (preparedStatement != null) {
+            try {
+                preparedStatement.close();
+            } catch (SQLException e) {
+                e.printStackTrace();
```

Comment:
printStackTrace isn't the best option of handling the error. Because you simply can lose it, and you won't be able to fix it. It's better to let a developer know that something bad happened. I suggest you adding:

`throw new RuntimeException(e.getMessage(), e);`